### ISSUE TITLE
test on aws

### ISSUE THREAD

**Issue by Borchardt** [focolab/minimo#8]

**Comment by Borchardt** [comment]
Testing on AWS Lightsail using an Amazon Linux 2 instance with 2 GB RAM, 1 vCPU, 60 GB SSD.

Testing steps:

Launch app: launch and connect to instance, [install Docker](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/docker-basics.html), [install Compose](https://gist.github.com/npearce/6f3c7826c7499587f00957fee62f8ee9), download and unzip repo, create static IP and attach to instance, add firewall rules to instance allowing HTTPS	on port 443 and TCP on 27017,  run `HOSTNAME=<static ip> docker-compose up -d --build` from repo root on instance.

Use app: go to https://<static ip>, create new comments form, upload folder with comments, verify that metadata and data browsers are populated appropriately.

App launches and is accessible at static ip, but data/metadata upload step fails silently. Web app container logs show:

```
S3Error: Moved Permanently
    at getError (/usr/src/app/node_modules/minio/dist/main/transformers.js:138:15)
    at /usr/src/app/node_modules/minio/dist/main/transformers.js:158:14
    at DestroyableTransform._flush (/usr/src/app/node_modules/minio/dist/main/transformers.js:80:10)
    at DestroyableTransform.<anonymous> (/usr/src/app/node_modules/readable-stream/lib/_stream_transform.js:115:49)
    at Object.onceWrapper (events.js:420:28)
    at DestroyableTransform.emit (events.js:314:20)
    at prefinish (/usr/src/app/node_modules/readable-stream/lib/_stream_writable.js:501:12)
    at finishMaybe (/usr/src/app/node_modules/readable-stream/lib/_stream_writable.js:513:7)
    at endWritable (/usr/src/app/node_modules/readable-stream/lib/_stream_writable.js:521:3)
    at DestroyableTransform.Writable.end (/usr/src/app/node_modules/readable-stream/lib/_stream_writable.js:491:41)
    at IncomingMessage.onend (_stream_readable.js:683:10)
    at Object.onceWrapper (events.js:420:28)
    at IncomingMessage.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  code: 'MovedPermanently',
  amzRequestid: null,
  amzId2: null,
  amzBucketRegion: null
}
```

More investigation to come.

**Comment by Borchardt** [comment]
Update:

Looks like this is a networking or execution environment peculiarity. For whatever reason, when running in this environment, the MinIO client will fall over during presigned puts if it isn't populated with an AWS region. Setting the region as an environment variable (`AWS_REGION` or `AWS_DEFAULT_REGION`) on the Lightsail instance causes a different failure, but adding the line `region: 'us-east-1'` to the MinIO client constructor in `server.js` works.

Testing completed successfully after adding that line to `server.js`. We can probably add a fix to add/populate this line programmatically according to a Docker Compose variable. Leaving this issue open until we make a decision about whether or not to do that and, if applicable, until that fix is implemented.

**Comment by Borchardt** [comment]
Closed by #22.
